### DISCUSSION TITLE
Feature Requests and Issue: Content Copying & Editor Hooks

### DISCUSSION THREAD

**Discussion by SatyaNarayana777** [yoopta-editor/Yoopta-Editor#413] [General]
I have encountered a few issues and potential feature requirements while using the editor. Below are the points for discussion:

Content Copying Indentation Issue:
When copying content from the editor to an external environment (e.g., a text editor), the indentation for numbered lists, bulleted lists, and checkboxes is not retained. This makes the copied content lose its intended structure. Can this behavior be improved to maintain formatting consistency during copy-paste actions?

Finding Selected Content:
Is there an existing function or hook that allows us to determine the content selected by the user within the editor? This would be helpful for scenarios where selection-based actions need to be implemented.

Adding Elements at Specific Positions:
Is there any function or hook available to programmatically insert elements (e.g., text, lists, or custom components) at a particular position within the editor? This would enhance the flexibility of customization and dynamic content generation.

Looking forward to insights or updates regarding these features!

**Comment by Darginec05** [comment]
Thank you @SatyaNarayana777 
These are really very good point! 

1. It's already planned for next release. Done by @gloaysa
2. Good point! I'll add it
3. Also in progress. It will be possible using Elements API

**Reply by SatyaNarayana777** [reply]
Thank you for the updates!
Excited to see these enhancements in the upcoming release. Looking forward to it!

**Comment by SatyaNarayana777** [comment]


**Reply by gloaysa** [reply]
As a rule of thumb, even if you want to develop the features yourself, I wouldn't trust the release planning of a project where the people work for free when they possibly can, and is bad etiquette asking about it.
